libraries/Network/src/NetworkManager.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "NetworkManager.h"
+#include "NetworkInterface.h"
+#include "esp_netif.h"
+#include "lwip/ip_addr.h"
+#include "lwip/dns.h"
+#include "esp32-hal-log.h"
+#include "esp_mac.h"
+
+NetworkManager::NetworkManager(){
+
+}
+
+bool NetworkManager::begin(){
+    static bool initialized = false;
+    if(!initialized){
+        initialized = true;
```

Comment:
I don't think this has any effect
```suggestion
```

Reply:
it has in the case where two tasks from separate cores decide to call `Network.begin()`. Not the best solution, but does well so far :)